lib/constants.go

```diff
@@ -50,6 +50,19 @@ const (
 	NodeErase
 )
 
+// Time constants
+const (
+	NanoSecondsPerSecond = uint64(1000000000)
+)
+
+func SecondsToNanoSeconds(secs uint64) uint64 {
+	return secs * NanoSecondsPerSecond
+}
+
+func NanoSecondsToSeconds(nanos uint64) uint64 {
+	return nanos / NanoSecondsPerSecond
```

Comment:
Careful about overflow. Just triple-check that overflow is impossible in all places where this is used.

Reply:
Copying response from slack to consolidate discussion here:

I checked all call-sites for this function and verified that every legacy callsite sets the previous TstampSecs to "seconds since Jan 1 1970". What's nice about that value is that it will always cleanly scale by 1e9 without overflowing. The only scenario where we get an overflow is if a caller passes in a value that's orders of magnitude higher than "seconds since Jan 1 1970", in which case the timestamp is wrong by 300+ years and is invalid anyway. It would fail block header validation. So, you're correct that we'll never hit that issue in practice. If we're REALLY concerned about it, what we can do is use a uint256 for the new TstampNanosSecs field so that it doesn't overflows when scaled by 1e9 in-memory (this has its own trade-offs btw). Viable solution though. I'm personally fine punting on adding this workaround until we see a block in the wild with such an invalid timestamp that fails to decode and scale properly. If one exists, we're guaranteed to see it when syncing a node with the latest code long before we ship PoS.